### ISSUE TITLE
Standards should be 'calculated'  instead of being provided by users

### ISSUE THREAD

**Issue by lock9** [neo-project/neo#2125]
**Summary or problem description**
We have smart contracts that affirm that they implement a certain standard, but the implemented methods differ in practice. For example, balanceOf, instead of returning an integer, returns a byte array. This breaks infra-structure tools that expect the standard to be followed. Smart contracts may 'break' too if they expect a type and get another instead. It is worth saying that interfaces enforce method signatures in any programming language (must be identical to the one in the interface). 
**TL;DR: Some smart contracts are not following the standards correctly. How can we avoid that?**

**Do you have any solution you want to propose?**
The standards should be 'calculated' when the smart contract is deployed to the network instead of being provided by the user (checking the ABI). 

**Neo Version**
- Neo 3

**Where in the software does this update applies to?**
- Ledger
- Network Policy

**Comment by roman-khimov** [comment]
See #1883 also.

**Comment by lock9** [comment]
@roman-khimov I will close this one and support your issue. Thanks!
